### PR TITLE
Make "make analyze" happy

### PR DESCRIPTION
**Pull request by siying**
Summary: "make analyze" is reporting some errors. It's complicated to look but it seems to me that they are all false positive. Anyway, I think cleaning them up is a good idea. Some of the changes are hacky but I don't know a better way.

Test Plan: Run "make analyze"

### CODE REVIEW

File: db/db_impl_write.cc [hunk 1]
```diff
@@ -247,6 +247,7 @@ Status DBImpl::WriteImpl(const WriteOptions& write_options,
         last_sequence = versions_->FetchAddLastToBeWrittenSequence(total_count);
       }
     }
+    assert(last_sequence != kMaxSequenceNumber);
```
Comment (siying) [inline db/db_impl_write.cc]: The warning was that in line 251, last_sequence  can be uninitialized. I didn't see it possible but I initialized it and added an assert anyway.

File: db/db_iter.cc [hunk 2]
```diff
@@ -644,6 +644,7 @@ void DBIter::ReverseToBackward() {
     while (iter_->Valid() &&
            user_comparator_->Compare(ikey.user_key, saved_key_.GetUserKey()) >
                0) {
+      assert(ikey.sequence != kMaxSequenceNumber);
       if (ikey.sequence > sequence_) {
```
Comment (siying) [inline db/db_iter.cc]: The warning was that in line 648 ikey.sequence can be unitialized. I hacked it to always initialize it to make the tool happy.

File: utilities/blob_db/blob_db_test.cc [hunk 3]
```diff
@@ -35,17 +35,7 @@ class BlobDBTest : public testing::Test {
...
-  }
-
-  void Reopen(BlobDBOptionsImpl bdb_options = BlobDBOptionsImpl(),
-              Options options = Options()) {
-    if (blob_db_) {
-      delete blob_db_;
-      blob_db_ = nullptr;
-    }
     ASSERT_OK(BlobDB::Open(options, bdb_options, dbname_, &blob_db_));
-    ASSERT_TRUE(blob_db_);
```
Comment (siying) [inline utilities/blob_db/blob_db_test.cc]: I spent two hours on this thing. The complain was that blob_db_ can be null but we called some functions. I tried all the way to hack the code to suppress the warning and the only way I made it work is this: not to support close and open (which we don't seem to use for now). By the way, there seems to be some memory leak cases in BlobDB::Open() in failure cases.

Comment (yiwu-arbug) [inline utilities/blob_db/blob_db_test.cc]: sounds good.

**Pull request closed without merging**